src/android/io/sentry/SentryCordova.java

```diff
@@ -1,47 +1,368 @@
+      crash();
+      return true;
+    }
+
+    try {
+      switch (action) {
+      case "startWithOptions":
+        JSONObject jsonOptions = args.getJSONObject(0);
+
+        startWithOptions(jsonOptions, callbackContext);
+
+        break;
+      case "captureEnvelope":
+        String envelope = args.getString(0);
+
+        captureEnvelope(envelope, callbackContext);
+
+        break;
+      case "setUser":
+        JSONObject jsonUser = null;
+        JSONObject otherUserKeys = null;
+
+        if (!args.isNull(0)) {
+          jsonUser = args.getJSONObject(0);
+        }
+        if (!args.isNull(1)) {
+          otherUserKeys = args.getJSONObject(1);
+        }
+
+        setUser(jsonUser, otherUserKeys, callbackContext);
+
+        break;
+      case "addBreadcrumb":
+        JSONObject jsonBreadcrumb = args.getJSONObject(0);
+
+        addBreadcrumb(jsonBreadcrumb, callbackContext);
+
+        break;
+      case "clearBreadcrumbs":
+        clearBreadcrumbs(callbackContext);
+        break;
+      case "getStringBytesLength":
+        String payload = args.getString(0);
+
+        int length = getStringBytesLength(payload);
+
+        callbackContext.sendPluginResult(new PluginResult(Status.OK, length));
+
+        break;
+      case "setTag":
+        String tagKey = args.getString(0);
+        String tag = args.getString(1);
+
+        setTag(tagKey, tag, callbackContext);
+
+        break;
+      case "setContext":
+        String contextKey = args.getString(0);
+        JSONObject contextObject = null;
+        if (!args.isNull(1)) {
+          contextObject = args.getJSONObject(1);
+        }
+
+        setContext(contextKey, contextObject, callbackContext);
+
+        break;
+      case "setExtra":
+        String extraKey = args.getString(0);
+        String extra = args.getString(1);
+
+        setExtra(extraKey, extra, callbackContext);
+
+        break;
+      default:
+        callbackContext.sendPluginResult(new PluginResult(Status.ERROR, "not implemented"));
+        break;
+      }
+
+    } catch (JSONException e) {
+      logger.info("Error parsing JSON from native bridge");
+
+      callbackContext.sendPluginResult(new PluginResult(Status.ERROR, false));
+
+      return false;
+    } catch (Exception e) {
+
+      callbackContext.sendPluginResult(new PluginResult(Status.ERROR, false));
+
+      return false;
+    }
+
+    return true;
+  }
+
+  private void startWithOptions(final JSONObject jsonOptions, final CallbackContext callbackContext) {
+    SentryAndroid.init(this.cordova.getActivity().getApplicationContext(), options -> {
+      try {
+        if (jsonOptions.has("dsn") && jsonOptions.getString("dsn") != null) {
           String dsn = jsonOptions.getString("dsn");
+          logger.info(String.format("Starting with DSN: '%s'", dsn));
+          options.setDsn(dsn);
+        } else {
+          // SentryAndroid needs an empty string fallback for the dsn.
+          options.setDsn("");
```

Comment:
the other option would be not even initing the SDK, checking before calling `SentryAndroid.init` if the Cordova allows a `null/invalid` DSN